src/components/ingestion/__test__/AbstractIngestionEngine.unit.test.js

```diff
@@ -0,0 +1,212 @@
+				expect(ingestionEngine.producer.send).toHaveBeenCalledTimes(1)
+			})
+		})
+
+		it('should send payloads to the correct topic', async () => {
+			jest.clearAllMocks()
+			const ingestionEngine = new FullyImplementedIngestionEngine()
+			ingestionEngine.producer = { send: jest.fn() }
+			const buffer = fs.readFileSync(path.join(__dirname, '/data/thinkingface.png'))
+			const payload = new Payload({ data: buffer })
+			const objectPassedToWriteable = stream.Writable.mock.calls[0][0]
+			const generatedWriteHandler = objectPassedToWriteable.write
+			await generatedWriteHandler(payload, null, () => {
+				const objectSentToKafka = ingestionEngine.producer.send.mock.calls[0][0]
+				expect(objectSentToKafka).toHaveProperty('topic')
+				expect(objectSentToKafka.topic).toEqual(dataTypes.STREAM.CONTAINER)
+			})
+		})
+
+		it('should send a deserializable Payload', async () => {
+			jest.clearAllMocks()
+			const ingestionEngine = new FullyImplementedIngestionEngine()
+			ingestionEngine.producer = { send: jest.fn() }
+			const buffer = fs.readFileSync(path.join(__dirname, '/data/thinkingface.png'))
+			const payload = new Payload({ data: buffer })
+			const objectPassedToWriteable = stream.Writable.mock.calls[0][0]
+			const generatedWriteHandler = objectPassedToWriteable.write
+			await generatedWriteHandler(payload, null, () => {
+				const objectSentToKafka = ingestionEngine.producer.send.mock.calls[0][0]
+				expect(objectSentToKafka.messages).toHaveLength(1)
+				const messageValue = objectSentToKafka.messages[0].value
+				expect(Payload.deserialize(messageValue)).toEqual(payload)
+			})
+		})
+	})
+
+	describe('processDemuxedPacket', () => {
+		it('should register new packets as most recent', () => {
+			const demuxedPacket = {
+				data: [
+					0, 0, 1, 0, 0, 215, 255, 251, 128, 0, 0, 1,
+				],
+				pts: 1255128,
+				dts: 1252125,
+				frame_ticks: 3003,
+				program: 1,
+				stream_number: 1,
+				stream_id: 224,
+				type: 1,
+				content_type: 2,
+				frame_num: 377,
+			}
+			const demuxedPacket2 = {
+				data: [
+					0, 0, 1, 0, 0, 215, 255, 251, 128, 0, 0, 1,
+				],
+				pts: 1255128,
+				dts: 1252125,
+				frame_ticks: 3003,
+				program: 1,
+				stream_number: 1,
+				stream_id: 224,
+				type: 1,
+				content_type: 2,
+				frame_num: 378,
+			}
+			const ingestionEngine = new FullyImplementedIngestionEngine()
+			ingestionEngine.processDemuxedPacket(demuxedPacket)
+			ingestionEngine.processDemuxedPacket(demuxedPacket2)
+			expect(ingestionEngine.mostRecentDemuxedPacket).toEqual(demuxedPacket2)
+		})
+	})
+
+	describe('getMostRecentDemuxedPacket', () => {
+		it('should return the value in most recent demuxed packet', () => {
+			const demuxedPacket = {
+				data: [
+					0, 0, 1, 0, 0, 215, 255, 251, 128, 0, 0, 1,
+				],
+				pts: 1255128,
+				dts: 1252125,
+				frame_ticks: 3003,
+				program: 1,
+				stream_number: 1,
+				stream_id: 224,
+				type: 1,
+				content_type: 2,
+				frame_num: 377,
+			}
```

Comment:
These sample packets are pretty heavy to have inline, does it violate anything to move them to a `/data/` file? Maybe then import them atop the file (as like `sampleDemuxedPackets`) and here do something like `const demuxedPacket = Object.assign({}, sampleDemuxedPackets[0])`?

Reply:
I think that would be very appropriate, yeah